java/clients/src/test/java/sleeper/clients/admin/AdminConfigStoreIT.java

```diff
@@ -188,10 +176,192 @@ void shouldRemoveDeletedTableFromLocalDirectoryWhenTablePropertyIsUpdated() thro
         }
     }
 
+    @DisplayName("Deploy instance property change with CDK")
+    @Nested
+    class DeployWithCdk {
+        @Test
+        void shouldRunCdkDeployWithLocalPropertiesFilesWhenCdkFlaggedInstancePropertyUpdated() throws Exception {
+            // Given
+            createTableInS3("test-table");
+            AtomicReference<InstanceProperties> localPropertiesWhenCdkDeployed = new AtomicReference<>();
+            List<TableProperties> localTablesWhenCdkDeployed = new ArrayList<>();
+            rememberLocalPropertiesWhenCdkDeployed(localPropertiesWhenCdkDeployed, localTablesWhenCdkDeployed);
+
+            // When
+            store().updateInstanceProperty(INSTANCE_ID, TASK_RUNNER_LAMBDA_MEMORY_IN_MB, "123");
+
+            // Then
+            verify(cdk).deploy();
+            assertThat(localPropertiesWhenCdkDeployed.get().get(TASK_RUNNER_LAMBDA_MEMORY_IN_MB))
+                    .isEqualTo("123");
+            assertThat(localTablesWhenCdkDeployed)
+                    .extracting(table -> table.get(TABLE_NAME))
+                    .containsExactly("test-table");
+        }
+
+        @Test
+        void shouldNotRunCdkDeployWhenUnflaggedInstancePropertyUpdated() {
+            // When
+            store().updateInstanceProperty(INSTANCE_ID, FARGATE_VERSION, "1.2.3");
+
+            // Then
+            verifyNoInteractions(cdk);
+        }
+
+        @Test
+        void shouldLeaveCdkToUpdateS3WhenApplyingChangeWithCdk() throws Exception {
+            // Given
+            instanceProperties.set(TASK_RUNNER_LAMBDA_MEMORY_IN_MB, "123");
+            instanceProperties.saveToS3(s3);
+
+            // When
+            store().updateInstanceProperty(INSTANCE_ID, TASK_RUNNER_LAMBDA_MEMORY_IN_MB, "456");
+
+            // Then
+            verify(cdk).deploy();
+            assertThat(store().loadInstanceProperties(INSTANCE_ID)
+                    .get(TASK_RUNNER_LAMBDA_MEMORY_IN_MB))
+                    .isEqualTo("123");
+        }
+
+        @Test
+        void shouldFailWhenCdkDeployFails() throws Exception {
+            // Given
+            IOException thrown = new IOException("CDK failed");
+            doThrow(thrown).when(cdk).deploy();
```

Comment:
I think it would be more consistent if these 2 lines are merged into one like the test below